meta-balena-common/recipes-core/initrdscripts/files/cryptsetup

```diff
@@ -2,13 +2,21 @@
 
 # shellcheck disable=SC1091
 . /usr/libexec/os-helpers-logging
+. /usr/libexec/os-helpers-tpm2
+
+EFI_DEV="/dev/disk/by-partlabel/balena-efi"
```

Comment:
Could we use `/dev/disk/by-state` instead? Any other link can suffer from clashes if a separate image with the same partition label if present. `by-state` will only link those partitions that are part of the boot media.

There is a `get_state_path_from_label()` function in `os-helpers-fs` that should preferably be used as it fallbacks to `by-label` and `by-partlabel`.

Reply:
On this front, there is also `balena-efi.service` that should use `by-state`:
```
ConditionPathExists=/dev/disk/by-label/balena-efi
```

Reply:
Fixed, together with the `by-state` symlink not being generated in the first place.